### DISCUSSION TITLE
GitHub App with opt-in permissions

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [community/community#31605] [Other Feature Feedback, Questions, & Ideas]
Is there a way for a GitHub permissions to be toggled on/off on a per-installation basis?

For example repo write permissions is something we don't need for core functionality but with it we can enhance our product experience. Yet some users would prefer to trade this the other way, no write permission for less functionality in the product.

AFAIK this is not possible, but I'd love to be wrong...

I suppose we could create multiple GitHub Apps but that solution is very coarse grained and adds a lot of complexity.

**Comment by MichaelKetting** [comment]
I've just got an update from the Jira app for Github that they now also want write access for "Content" so branches can be created. That's a feature I don't need and I don't want to grant full write access to my repository.

I would love the option to deny certain permissions and change the list of permissions that have been granted to a Github app.

**Reply by doublerebel** [reply]
Filed as #38382

**Comment by doublerebel** [comment]
Related: #15733
